### PR TITLE
Feature/graph helper

### PR DESCRIPTION
**Pull request by Edds390**
Related to issue  [13](https://github.com/Edds390/solarnetwork-aggregated-dashboard/issues/13) and [6](https://github.com/Edds390/solarnetwork-aggregated-dashboard/issues/6).

Features

- Implemented filtering feature, where the graph will update based on the different values of the checkbox. The checkbox interface has yet to be implemented, but I've made a datastructure that is a JSON object with key as the node-data source combination and the value as a boolean flag that represents whether that combination should be displayed in the graph.

- Parser can now parse data sources from multiple nodes. The parser will concatenate the Node ID and the data source as the label (e.g. Node183 Solar)

- Promoted DashboardPanel into a smart component that holds the state of data and passes any model data into the graphs for parsing.

- Added Switch for changing the stack view of the graph.

- Changed presentation layout of the graph and legend.

- Added a popover for displaying help on how to interact with the graph.

Dependencies Added

- Material-ui-next Grid: for responsive design. Our version of material-ui did not have the grid component so I had to use yarn to install the other version as well as keep our old one.
- Yarn: to allow for the material-ui-next and material-ui libraries to be in the same project.

### CODE REVIEW

**Comment by SeoyoungC** [comment]
Running into an error when running 'npm install' - does not seem compatible with npm:
screen shot 2018-03-27 at 3 35 23 pm

**Comment by Edds390** [comment]
Material UI Grid needed a yarn dependency to install and I am aware that not all dev platforms have yarn installed. I have replaced the Grid with react-bootstrap which accomplishes the same thing without the need for the yarn installation.

**Comment by SeoyoungC** [comment]
Found visual issue with table and graph - overlapping:
screen shot 2018-03-27 at 4 26 55 pm
Not a huge problem now, formatting can be dealt with after main components have been merged in!

File: src/components/MasterChart/MasterChart.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+import Toggle from 'material-ui/Toggle';
 import Dygraph from 'dygraphs';
+import _ from 'lodash';
 import DataParser from '../../utils/DataParser';
+import GraphTutorialPopover from '../GraphTutorialPopover/GraphTutorialPopover';
 import nodeInfo from '../../utils/Data/nodeInfo';
 import nodeInfo205 from '../../utils/Data/nodeInfo205';
+import './MasterChart.css';
+
+const GRAPH_HEIGHT = 320;
```
Comment (SeoyoungC) [inline src/components/MasterChart/MasterChart.js:14]: May need to look at this value, and see if we can set graph height dynamically to cater for all screen sizes/resolutions.

**Review by SeoyoungC — APPROVED** [review]
Other than small visual changes that we can address later, everything looks good!

**Review by elisalter — APPROVED** [review]
This looks good! We should think about adding CSS loaders eventually, as the current method of dumping link tags in `index.html` is not maintainable. However, to use CSS loaders we will need to eject from create-react-app and change webpack config.

**Pull request merged**